airflow/providers/amazon/aws/triggers/glue_crawler.py

```diff
@@ -0,0 +1,78 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import asyncio
+from functools import cached_property
+from typing import AsyncIterator
+
+from botocore.exceptions import WaiterError
+
+from airflow.providers.amazon.aws.hooks.glue_crawler import GlueCrawlerHook
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+
+class GlueCrawlerCompleteTrigger(BaseTrigger):
+    """
+    Watches for a glue crawl, triggers when it finishes
+
+    :param crawler_name: name of the crawler to watch
+    :param poll_interval: The amount of time in seconds to wait between attempts.
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    """
+
+    def __init__(self, crawler_name: str, poll_interval: int, aws_conn_id: str):
+        super().__init__()
+        self.crawler_name = crawler_name
+        self.poll_interval = poll_interval
+        self.aws_conn_id = aws_conn_id
+
```

Comment:
This applies to both Triggers, but is there something that is tracking the number of attempts the Trigger makes? The `timeout` is not set, which means that there is no timeout. What is the exit criteria if the service is indefinitely down?

Reply:
there was no limit to the number of tries if `wait_for_completion` before, so I kept the behavior 🤷